### PR TITLE
Add controller config to the cache controller.

### PR DESCRIPTION
**Pull request by howbazaar**
This branch adds controller config into the cache.

The branch also introduces additional change mechanisms to the modelcache worker. The modelcache worker is the thing that is responsible for populating the model cache. This worker now also listens to pubsub messages for updates. Just now it only listens to the controller config changed events, but it opens up for future additions.

This also introduces a finer grained change to the cache controller.

The controller name is only used now for the model summary watcher.

## QA steps

```sh
juju bootstrap
check the juju_engine_report to ensure workers started as expected
```

### CODE REVIEW

**Comment by howbazaar** [comment]
Check found an intermittent test failure worth noting:

```
---------------------------------------------------------------------
FAIL: txnwatcher_test.go:284: TxnWatcherSuite.TestDoubleUpdate

[LOG] 0:00.019 TRACE test loop started
insert("test", "a") => revno 2
update("test", "a") => revno 3
[LOG] 0:00.049 TRACE test txn watcher 0xc000477040 starting sync
[LOG] 0:00.050 TRACE test 0xc000477040 step 1 got changelog document: bson.D{bson.DocElem{Name:"_id", Value:"^T\xb3\xa1\xb2==\x1a\xbb\x96%]"}, bson.DocElem{Name:"test", Value:bson.D{bson.DocElem{Name:"d", Value:[]interface {}{"a"}}, bson.DocElem{Name:"r", Value:[]interface {}{3}}}}}
[LOG] 0:00.050 TRACE test 0xc000477040 step 2 got changelog document: bson.D{bson.DocElem{Name:"_id", Value:"^T\xb3\xa0\xb2==\x1a\xbb\x96%\\"}, bson.DocElem{Name:"test", Value:bson.D{bson.DocElem{Name:"d", Value:[]interface {}{"a"}}, bson.DocElem{Name:"r", Value:[]interface {}{2}}}}}
update("test", "a") => revno 4
[LOG] 0:00.052 TRACE test txn watcher 0xc000477040 starting sync
[LOG] 0:00.052 TRACE test 0xc000477040 step 3 got changelog document: bson.D{bson.DocElem{Name:"_id", Value:"^T\xb3\xa1\xb2==\x1a\xbb\x96%^"}, bson.DocElem{Name:"test", Value:bson.D{bson.DocElem{Name:"d", Value:[]interface {}{"a"}}, bson.DocElem{Name:"r", Value:[]interface {}{4}}}}}
txnwatcher_test.go:295:
    c.Assert(hub.values, jc.DeepEquals, []watcher.Change{
        {"test", "a", revno1},
        {"test", "a", revno3},
    })
... obtained []watcher.Change = []watcher.Change{watcher.Change{C:"test", Id:"a", Revno:3}, watcher.Change{C:"test", Id:"a", Revno:4}}
... expected []watcher.Change = []watcher.Change{watcher.Change{C:"test", Id:"a", Revno:2}, watcher.Change{C:"test", Id:"a", Revno:4}}
... mismatch at [0].Revno: unequal; obtained 3; expected 2

[LOG] 0:00.069 TRACE test loop finished
```

**Comment by howbazaar** [comment]
!!build!!

File: worker/modelcache/manifold_test.go [hunk 1]
```diff
@@ -146,3 +221,24 @@ type fakeWorker struct {
...
+}
+
+// Return an invalid but non-zero state pool pointer.
+// Is only ever used for comparison.
+func (f *fakeStateTracker) Use() (*state.StatePool, error) {
+	return f.pool(), nil
+}
+
+func (f *fakeStateTracker) pool() *state.StatePool {
+	return (*state.StatePool)(unsafe.Pointer(f))
```
Comment (jameinel) [inline worker/modelcache/manifold_test.go]: Can we do this instead of unsafe?
```
return &state.StatePool{}
```

**Comment by howbazaar** [comment]
$$merge$$

**Comment by howbazaar** [comment]
$$merge$$

**Pull request merged**